sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -85,14 +86,16 @@ case class AdaptiveSparkPlanExec(
   @transient private val removeRedundantProjects = RemoveRedundantProjects
   @transient private val removeRedundantSorts = RemoveRedundantSorts
   @transient private val ensureRequirements = EnsureRequirements
+  @transient private val disableUnnecessaryBucketedScan = DisableUnnecessaryBucketedScan
 
   // A list of physical plan rules to be applied before creation of query stages. The physical
   // plan should reach a final status of query stages (i.e., no more addition or removal of
   // Exchange nodes) after running these rules.
   private def queryStagePreparationRules: Seq[Rule[SparkPlan]] = Seq(
     removeRedundantProjects,
     removeRedundantSorts,
-    ensureRequirements
+    ensureRequirements,
+    disableUnnecessaryBucketedScan
```

Comment:
seems we can just put `DisableUnnecessaryBucketedScan`, same to other rules.

Reply:
@cloud-fan - agree, updated.